### PR TITLE
chore: Remove namespace restriction for EnvoyProxy parametersRef reso…

### PR DESCRIPTION
**Pull request by phantooom**
**What type of PR is this?**
* Remove namespace restriction for EnvoyProxy parametersRef 

FIX: https://github.com/envoyproxy/gateway/issues/3528

### CODE REVIEW

**Review by cnvergence — APPROVED** [review]
thanks

**Review by zirain — APPROVED** [review]

File: internal/provider/kubernetes/controller.go [hunk 1]
```diff
@@ -1578,14 +1578,14 @@ func (r *gatewayAPIReconciler) processParamsRef(ctx context.Context, gc *gwapiv1
 	}
 
 	epList := new(egv1a1.EnvoyProxyList)
+	gcParametersRefNamespace := string(*gc.Spec.ParametersRef.Namespace)
```
Comment (arkodg) [inline internal/provider/kubernetes/controller.go:1581]: nil check ?

Comment (arkodg) [reply internal/provider/kubernetes/controller.go:1581]: nvm its checked in `refsEnvoyProxy` https://github.com/envoyproxy/gateway/blob/516a27d6d7d8d35686fc7b8391864b123aaea7f7/internal/provider/kubernetes/helpers.go#L195

**Review by arkodg — APPROVED** [review]
thanks !

**Pull request merged**